BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -2030,5 +1893,57 @@ private float AdjustValue(float min, float max, float value, float step)
 
             return adjusted_value;
         }
+
+        private Device GetDevice()
+        {
+            Device device = _context.Devices[0];
+
+            if (string.IsNullOrEmpty(this.SerialNumber))
+                return device;
+
+            foreach(Device dev in _context.Devices)
+            {
+                if (dev.Info[CameraInfo.SerialNumber] == this.SerialNumber)
+                {
+                    device = dev;
+                    break;
+                }
+            }
+
+            return device;
+        }
+
+        private Sensor GetSensor(string sensorName)
+        {
+            Device dev = GetDevice();
+            Sensor s = dev.Sensors[0];
+
+            foreach(Sensor sen in dev.Sensors)
+            {
+                Sensor.CameraInfos infos = sen.Info;
+                string info = infos[CameraInfo.Name];
+                if(info == sensorName)
+                {
+                    s = sen;
+                    break;
+                }
+            }
+
+            return s;
+        }
+
+        private List<Point2i> GetSupportedResolutions(string sensorName)
+        {
+            return GetSensor(sensorName).StreamProfiles.Select(p => new Point2i()
```

Comment:
You could use `Sensor.VideoStreamProfiles` to get rid of the `as VideoStreamProfile` casts below.